pkg/core/hooks/conn/util.go

```diff
@@ -167,6 +167,12 @@ func Capture(_ context.Context, logger *zap.Logger, t chan *models.TestCase, req
 		return
 	}
 
+	reqBody, errDecompress := pkg.Decompress(logger, req.Header.Get("Content-Encoding"), reqBody)
```

Comment:
Add a condition here that if value of "Content-Encoding" is not empty then only Call that function, better for readability.